src/classes/VOL_CTRL_SendBulkEmail.cls

```diff
@@ -118,28 +118,112 @@ public with sharing class VOL_CTRL_SendBulkEmail {
+    * @description Sets the email body and subject from the template matching the template Id selected
+    * @return null
+    */
+    public PageReference loadEmailTemplate() {
+        strSubject = null;
+        strMessage = null;
+
+        if (templateId == null) {
+            return null;
+        }
+
         try {
-            if (templateId != null) {
-                // Ensure the user has access to the object and fields before querying
-                UTIL_Describe.checkReadAccess('EmailTemplate', new Set<String>{'Id','Name', 'Subject', 'HtmlValue', 'Body'});    
-                    
-                EmailTemplate et = [select Id, Name, Subject, HtmlValue, Body from EmailTemplate where Id = :templateId];
-                strSubject = et.Subject;
-                strMessage = (et.HtmlValue != null) ? et.HtmlValue : et.Body;
+            // Ensure the user has access to the object and fields before querying
+            UTIL_Describe.checkReadAccess(
+                'EmailTemplate',
+                new Set<String>{ 'Id', 'Name', 'Subject', 'HtmlValue', 'Body' }
+            );
+
+            EmailTemplate template = [
+                SELECT Id, Name, Subject, HtmlValue, Body, TemplateType
+                FROM EmailTemplate
+                WHERE Id = :templateId
+            ];
+
+            strSubject = template.Subject;
+
+            if (template.HtmlValue != null) {
+                strMessage = cleanHtml(template);
+
             } else {
-                strSubject = null;
-                strMessage = null;      
+                strMessage = template.Body;
             }
+
             return null;
+
         } catch (Exception e) {
-            ApexPages.addMessages(e);  
+            ApexPages.addMessages(e);
             return null;
         }
     }
 
+    /*******************************************************************************************************************
+    * @description Html values are only cleaned automatically by Salesforce when the TemplateType is html
+    * in order to clean the other html values, we will do a temporary commit to a Rich Text Area field to allow
+    * Salesforce to strip the non whitelisted tags for us
+    * @return Volunteer_Job__c the job associated with this page
+    */
+    @TestVisible
+    private String cleanHtml(EmailTemplate template) {
+        if (template.TemplateType.equalsIgnoreCase('html')) {
+            return template.HtmlValue;
+        }
+
+        Savepoint savepoint = Database.setSavePoint();
+        String value = '';
+        Volunteer_Job__c job = getRelatedJob();
+
+        try {
+            job.Description__c = template.HtmlValue;
+            upsert job;
+            value = [SELECT Description__c FROM Volunteer_Job__c WHERE Id = :job.Id LIMIT 1].Description__c;
+
+            // Intentionally throw an error to ensure a rollback always occurs even when a unintentional error occurs
+            throw new SendBulkEmailException('');
```

Comment:
Since you want the rollback to run whether or not there was an exception in the `try` block, you can use a try-finally block with no catch to avoid the need for thowing this exception. i.e.

```
try {
    job.Description__c = template.HtmlValue;
    upsert job;
    value = [SELECT Description__c FROM Volunteer_Job__c WHERE Id = :job.Id LIMIT 1].Description__c;
} finally {
    Database.rollback(savepoint);
}
```

This also has the advantage that if there's an unexpected exception, it won't get accidentally ignored.

Reply:
Oh yea, much cleaner approach, thanks!